components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -96,10 +96,9 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx)
             //throw new IdentityOAuth2Exception("Error when authenticating the user credentials", e);
         }
         if (authStatus) {
-            if (username.indexOf(CarbonConstants.DOMAIN_SEPARATOR) < 0) {
-                if (UserCoreUtil.getDomainFromThreadLocal() != null && !UserCoreUtil.getDomainFromThreadLocal().equals("")) {
-                    username = UserCoreUtil.getDomainFromThreadLocal() + CarbonConstants.DOMAIN_SEPARATOR + username;
-                }
+            if (username.indexOf(CarbonConstants.DOMAIN_SEPARATOR) < 0 && UserCoreUtil.getDomainFromThreadLocal() !=
+                    null && !"".equals(UserCoreUtil.getDomainFromThreadLocal())) {
```

Comment:
Use contains() instead of indexOf() and Use StringUtils.isNotEmpty for null safe empty check